internal/server/server.go

```diff
@@ -127,56 +128,31 @@ func Serve(
 		return err
 	}
 
-	baseMux := http.NewServeMux()
-	baseMux.Handle("/", gwmux)
+	httpMux := http.NewServeMux()
+	httpMux.Handle("/", gwmux)
 
-	httpServer := &http.Server{
-		Handler:      grpcHandlerFunc(grpcServer, baseMux),
-		Addr:         address,
-		ReadTimeout:  5 * time.Second,
-		WriteTimeout: 10 * time.Second,
-		IdleTimeout:  120 * time.Second,
+	logger.Info("server started")
+	if err := mux.Serve(
+		ctx,
+		mux.WithHTTPTarget(config.addr(), &http.Server{
+			Handler:      grpcHandlerFunc(grpcServer, httpMux),
```

Comment:
Why are we using `grpcHandlerFunc` here? We should be able to directly pass `gwmux` here.